src/formBuilder/formBuilder.jsx

```diff
@@ -241,6 +242,27 @@ const FormBuilder = props => {
     );
   };
 
+  useEffect(() => {
+    // Reset form fields in case these are updated
+    const prevFields = [].concat(...formFields);
+    const prevFieldNames = prevFields.map(f => f.name);
+    const currFields = [].concat(...fields).map(f => f.name);
+    if (!currFields.every(fName => prevFieldNames.includes(fName))) {
```

Comment:
I'm confused on this a bit. What is the instance when this can happen?

Reply:
this new feature is for cases when you switch forms from default to advanced mode. I was having issues when the No. of fields changes, so field values weren't being assigned properly in the form. Found that I needed to do a deep form reset

Reply:
I still see this happening when i toggle the safe minion launch form - i think we need to look into a new method here.

@jordanlesich has done some work on this.